runtimebp/cpu.go

```diff
@@ -127,11 +134,108 @@ func defaultMaxProcsFormula(n float64) int {
 	return int(math.Ceil(n))
 }
 
-// GOMAXPROCS sets runtime.GOMAXPROCS with the default formula,
+func scaledMaxProcsFormula(n float64) int {
+	// If we have a request below 1, round up so we don't end up single-threaded
+	if n < 1 {
+		n = 1
+	}
+	i := fetchMaxProcsMult()
+	return int(math.Ceil(n * float64(i)))
+}
+
+func fetchMaxProcsMult() int {
+	// Allow using a multiplier for number of processes relative to limit.
+	// Not catching the ok here because this function will never be called unless
+	// the environment variable is set
+	v, _ := os.LookupEnv("BASEPLATE_GOMAXPROCSMULT")
+	intVal, err := strconv.Atoi(v)
+	if err != nil {
+		fmt.Fprintf(
+			os.Stderr,
+			"runtimebp.fetchMaxProcsMult: GOMAXPROCSMULT is set to %q, which is invalid, falling back to %d to ensure parallelism: %v",
+			v,
+			defaultMultiplier,
+			err,
+		)
+		intVal = defaultMultiplier
+	}
+	if intVal <= 0 {
+		intVal = 1
```

Comment:
Shouldn't this also just fallback to the default multiplier 2 here since an invalid value was set by the environment?